scripts/backfill_source_tags_for_unregistered_contributors.py

```diff
@@ -0,0 +1,181 @@
+from datetime import timedelta
+import logging
+from django.apps import apps
+from website.app import init_app
+from scripts import utils as script_utils
+import datetime
+import progressbar
+from django.core.paginator import Paginator
+
+import sys
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def backfill_source_tags_for_osf4m_unregistered_contributors(dry_run):
+    """ Backfill osf4m source tags to all osf4m unregistered contributors
+    """
+    Tag = apps.get_model('osf', 'Tag')
+    OSFUser = apps.get_model('osf', 'OSFUser')
+    NodeLog = apps.get_model('osf', 'NodeLog')
+    meeting_source_tag = Tag.all_tags.get(name='source:campaign|osf4m', system=True)
+    meeting_nodes = meeting_source_tag.abstractnode_tagged.all()
+    logging.info('Number of meeting nodes found: ' + str(len(meeting_nodes)))
+    all_meeting_node_logs = NodeLog.objects.filter(action='contributor_added', node__in=meeting_nodes).only('created', 'params')
+    ThroughModel = OSFUser.tags.through
```

Comment:
Because `OSFUser` maintain a M2M relationship to `Tag`, for the sake of performance, we do a `bulk_create` directly on the through model, which populates the intermediary table used to maintain the M2M relationship.